### PR TITLE
TASK: Tweak composer scripts

### PR DESCRIPTION
**Pull request by bwaidelich**
* Move PHPCS configuration to `phpcs.xml.dist`
* Add script `lint:phpcs:fix` to allow fixing of PHPCS errors
* Reenable `PSR1.Files.SideEffects` PHPCS sniff (that was excluded with a864ced0eb0fd94e35a252efc9bdf923b021652f but that seems no longer to be necessary)
* Add scripts `test:behavioral:sync` and `test:behavioral:stop-on-failure:sync` to centralize `CATCHUPTRIGGER_ENABLE_SYNCHRONOUS_OPTION` hack

### CODE REVIEW

**Comment by bwaidelich** [comment]
draft because this removes the `lint:phpcs-psr12` script and I'm not sure whether @nezaniel still relies on it

File: .composer.json [hunk 1]
```diff
@@ -47,6 +45,10 @@
       "../../flow doctrine:migrate --quiet; ../../flow cr:setup",
       "@test:behat-cli -c Neos.Neos/Tests/Behavior/behat.yml"
     ],
+    "test:behavioral:sync": [
+      "@putenv CATCHUPTRIGGER_ENABLE_SYNCHRONOUS_OPTION=1",
```
Comment (mhsdesign) [inline .composer.json:49]: TIL about `@putenv` ^^

**Review by mhsdesign — APPROVED** [review]
Thanks :) Was somewhere on my todo list ^^

**Comment by ahaeslich** [comment]
Why did we merge this when the last pipeline didn't pass? 

Can we please be more cautious. Or at least leave a comment why this was ignored. It's pretty annoying to find this afterwards.

**Comment by mhsdesign** [comment]
The optional step

> Add Labels to Pull Request 

failed. (due to an api rate blablablabla)

**Pull request merged**